### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by AlexanderVeselov**
Fixed a lot of warnings!

### CODE REVIEW

File: Baikal/Renderers/monte_carlo_renderer.cpp [hunk 1]
```diff
@@ -31,7 +31,7 @@
 #include <random>
 #include <algorithm>
 
-#include "Utils/sobol.h"
+//#include "Utils/sobol.h"
```
Comment (Kvalme) [inline Baikal/Renderers/monte_carlo_renderer.cpp]: remove

File: Baikal/Utils/cl_inputmap_generator.cpp [hunk 2]
```diff
@@ -101,7 +101,7 @@ void CLInputMapGenerator::GenerateInputSource(std::shared_ptr<Baikal::InputMap>
 
         case InputMap::InputMapType::kConstantFloat:
         {
-            InputMap_ConstantFloat *i = static_cast<InputMap_ConstantFloat*>(input.get());
+            //InputMap_ConstantFloat *i = static_cast<InputMap_ConstantFloat*>(input.get());
```
Comment (Kvalme) [inline Baikal/Utils/cl_inputmap_generator.cpp]: remove dead code

File: BaikalIO/material_io.cpp [hunk 3]
```diff
@@ -227,13 +227,13 @@ namespace Baikal
...
-        material->LinkRefractionIOR(element.Attribute("refraction_link_ior") == "true");
-        material->SetMultiscatter(element.Attribute("sss_multyscatter") == "true");
+        material->SetDoubleSided(strcmp(element.Attribute("emission_doublesided"), "true") == 0);
+        material->LinkRefractionIOR(strcmp(element.Attribute("refraction_link_ior"), "true") == 0);
+        material->SetMultiscatter(strcmp(element.Attribute("sss_multyscatter"), "true") == 0);
         material->SetName(name);
 
         auto num_inputs = material->GetNumInputs();
-        for (size_t a = 0; a < num_inputs; ++a)
+        for (std::size_t a = 0u; a < num_inputs; ++a)
```
Comment (Kvalme) [inline BaikalIO/material_io.cpp:236]: why you add std:: before size_t?

File: BaikalStandalone/Application/application.h [hunk 4]
```diff
@@ -66,7 +66,7 @@ namespace Baikal
         int m_num_triangles;
         int m_num_instances;
 
-        int m_shape_id_val;
+        //int m_shape_id_val;
```
Comment (Kvalme) [inline BaikalStandalone/Application/application.h]: Instead of commenting out - just remove

File: BaikalTest/input_maps.h [hunk 5]
```diff
@@ -554,7 +554,7 @@ TEST_F(InputMapsTest, InputMap_Shuffle)
 {
     auto material = Baikal::UberV2Material::Create();
     auto color1 = Baikal::InputMap_ConstantFloat3::Create(float3(1.0f, 0.0f, 0.0f));
-    auto diffuse_color = Baikal::InputMap_Shuffle::Create(color1, {0, 0, 0, 0});
+    auto diffuse_color = Baikal::InputMap_Shuffle::Create(color1, {{0, 0, 0, 0}});
```
Comment (Kvalme) [inline BaikalTest/input_maps.h:557]: Why you made this change?

**Review by Kvalme — CHANGES_REQUESTED** [review]
I don't see changes that enables all warnings and make them errors.

File: BaikalTest/input_maps.h (hunk 5 above)
Comment (AlexanderVeselov) [reply BaikalTest/input_maps.h:557]: It seems to be a bug in clang! See [stackoverflow](https://stackoverflow.com/questions/31555584/why-is-clang-warning-suggest-braces-around-initialization-of-subobject-wmis).

Comment (Kvalme) [reply BaikalTest/input_maps.h:557]: then just move this part out to std::array<uint32_t, 4> shuffle = {0, 0, 0, 0};

Comment (AlexanderVeselov) [reply BaikalTest/input_maps.h:557]: Nothing has changed, i've got the same warning. Clang doesn't allow me to use initializer list with single braces anyway.

**Comment by Kvalme** [comment]
Please, revert std::size_t change to size_t.
Seems that we should use types from std namespace

**Pull request closed without merging**